jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/FileSystemPool.java

```diff
@@ -0,0 +1,273 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2022 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.util.resource;
+
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystemAlreadyExistsException;
+import java.nio.file.FileSystemNotFoundException;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.FileTime;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.eclipse.jetty.util.IO;
+import org.eclipse.jetty.util.thread.AutoLock;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class FileSystemPool
```

Comment:
Can this be a LifeCycle?  (perhaps tied to the Server?)

I think it would be useful to close open `FileSystem` objects.
As well as showing if we have a resource leak on stop.

Reply:
It could, but this needs to be very carefully thought out.
Mounted filesystems are global to the JVM so if we have two servers sharing the same `FileSystemPool` instance, or two `FileSystemPool` instances within a single JVM, things are going to turn awry.

Reply:
Who calls sweep is interesting..... Is that a lifecycle? Does each resident server call sweep?
Happy to not solve now and have TODO and/or issue for that